src/objects/domain_obj.f90

```diff
@@ -1657,8 +1662,20 @@ subroutine read_land_variables(this, options)
             if (allocated(this%land_mask)) then
                 this%land_mask = temporary_data(this%grid%ims:this%grid%ime, this%grid%jms:this%grid%jme)
             endif
+            where(this%land_mask==0) this%land_mask = kLC_WATER  ! To ensure conisitency. land_mask can be 0 or 2 for water, enforce a single value.
         endif
 
+        if ((options%physics%watersurface==kWATER_LAKE) .AND.(options%parameters%lakedepthvar /= "")) then
+            if (this_image()==1) write(*,*) "   reading lake depth data from hi-res file"
+
+            call io_read(options%parameters%init_conditions_file,   &
+                           options%parameters%lakedepthvar,         &
+                           temporary_data)
+            if (associated(this%lake_depth%data_2d)) then
+                this%lake_depth%data_2d = temporary_data(this%grid%ims:this%grid%ime, this%grid%jms:this%grid%jme)
+            endif
+
+        endif
```

Comment:
do we need an `if watersurface==kWATER_LAKE and lakedepthvar==""` and set a fixed value for the lake model depth in case someone doesn't have lake depth data? (though print a warning in that case!)

Reply:
The lake model does exactly that (in lakeini) if no lakedepth is provided